Marlin/Configuration.h

```diff
@@ -808,10 +808,21 @@
   #define ABL_GRID_POINTS_Y ABL_GRID_POINTS_X
 
   // Set the boundaries for probing (where the probe can reach).
-  #define LEFT_PROBE_BED_POSITION 15
-  #define RIGHT_PROBE_BED_POSITION 170
-  #define FRONT_PROBE_BED_POSITION 20
-  #define BACK_PROBE_BED_POSITION 170
+  #define PROBE_BED_MARGIN 5
+  #if X_PROBE_OFFSET_FROM_EXTRUDER < 0
+    #define LEFT_PROBE_BED_POSITION PROBE_BED_MARGIN
+    #define RIGHT_PROBE_BED_POSITION (X_MAX_POS - PROBE_BED_MARGIN + X_PROBE_OFFSET_FROM_EXTRUDER)
+  #else
+    #define LEFT_PROBE_BED_POSITION (X_PROBE_OFFSET_FROM_EXTRUDER + PROBE_BED_MARGIN)
+    #define RIGHT_PROBE_BED_POSITION (X_MAX_POS - PROBE_BED_MARGIN)
+  #endif
+  #if X_PROBE_OFFSET_FROM_EXTRUDER < 0
+    #define FRONT_PROBE_BED_POSITION PROBE_BED_MARGIN
+    #define BACK_PROBE_BED_POSITION (Y_MAX_POS - PROBE_BED_MARGIN + Y_PROBE_OFFSET_FROM_EXTRUDER)
+  #else
+    #define FRONT_PROBE_BED_POSITION (Y_PROBE_OFFSET_FROM_EXTRUDER + PROBE_BED_MARGIN)
+    #define BACK_PROBE_BED_POSITION (Y_MAX_POS - PROBE_BED_MARGIN)
+  #endif
```

Comment:
I've had mine this for a while now:

```
// Set the boundaries for probing (where the probe can reach). Calculations set to max dimensions inset by MIN_PROBE_EDGE. Replace calculations below with hard-coded values to set manually.
  #define LEFT_PROBE_BED_POSITION  (X_PROBE_OFFSET_FROM_EXTRUDER <= MIN_PROBE_EDGE ? MIN_PROBE_EDGE : X_PROBE_OFFSET_FROM_EXTRUDER) 
  #define RIGHT_PROBE_BED_POSITION (X_PROBE_OFFSET_FROM_EXTRUDER >= -MIN_PROBE_EDGE ? X_MAX_POS - MIN_PROBE_EDGE : X_MAX_POS + X_PROBE_OFFSET_FROM_EXTRUDER)
  #define FRONT_PROBE_BED_POSITION (Y_PROBE_OFFSET_FROM_EXTRUDER <= MIN_PROBE_EDGE ? MIN_PROBE_EDGE : Y_PROBE_OFFSET_FROM_EXTRUDER)
  #define BACK_PROBE_BED_POSITION (Y_PROBE_OFFSET_FROM_EXTRUDER >= -MIN_PROBE_EDGE ? Y_MAX_POS - MIN_PROBE_EDGE : Y_MAX_POS + Y_PROBE_OFFSET_FROM_EXTRUDER)
```

Submitting this simple PR in order to get my feet wet has been on my todo list for quite some time and haven't gotten to it, so please take if you think this works better.